lib/modules/eic_groups/src/Plugin/Block/SearchMenuGroupBlock.php

```
@@ -0,0 +1,32 @@
+<?php
+
+namespace Drupal\eic_groups\Plugin\Block;
+
+use Drupal\Core\Block\Annotation\Block;
+use Drupal\Core\Block\BlockBase;
+use Drupal\eic_groups\Form\SearchMenuGroupForm;
+
+/**
+ * Provides an SearchMenuGroupBlock block.
+ *
+ * @Block(
+ *   id = "eic_groups_search_menu_group",
+ *   admin_label = @Translation("EIC Search Menu Group"),
+ *   category = @Translation("European Innovation Council"),
+ *   context_definitions = {
+ *     "group" = @ContextDefinition("entity:group", required = FALSE)
+ *   }
+ * )
+ */
+class SearchMenuGroupBlock extends BlockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $form = \Drupal::formBuilder()->getForm(SearchMenuGroupForm::class);
```

Can you use dependency injection here instead of calling `\Drupal::formBuilder()`? Usually grumphp should throw an error when you try to use drupal services within a class without using proper dependency injection.
In order to avoid spending time searching about how to implement dependency injection in plugins, I already did that for you :):

- Your class should implement the `Drupal\Core\Plugin\ContainerFactoryPluginInterface`;
- Implement the following create method:
```php
/**
 * {@inheritdoc}
 */
public static function create(
  ContainerInterface $container,
  array $configuration,
  $plugin_id,
  $plugin_definition
) {
  return new static(
    $configuration,
    $plugin_id,
    $plugin_definition,
    $container->get('form_builder')
  );
}
```

- Override the constructor:

```php
public function __construct(
  array $configuration,
  $plugin_id,
  $plugin_definition,
  FormBuilderInterface $form_builder,
) {
  parent::__construct($configuration, $plugin_id, $plugin_definition);
  $this->formBuilder = $form_builder;
}
```

---

Haha thanks for doing my job 🍸 🥥